corehq/apps/app_manager/app_schemas/casedb_schema.py

```diff
@@ -43,9 +42,7 @@ def _add_ancestors(ctype, generation):
     subsets = [{
         "id": generation_names[i],
         "name": "{} ({})".format(generation_names[i], " or ".join(ctypes)) if i > 0 else base_case_type,
-        "structure": {
-            p: {"description": descriptions_dict.get(t, {}).get(p, '')}
-            for t in ctypes for p in map[t]},
+        "structure": generate_structure(ctypes, map, descriptions_dict),
```

Comment:
Is this change a just a refactor to make it more readable? Or is there a logic change as well?

Reply:
the change within `generate_property_metadata` needed to be made, so that was what encouraged the refactoring, but yes, this is essentially just refactoring.